pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -3185,39 +3185,51 @@ func TestReconcileCancelledPipelineRun(t *testing.T) {
 	}
 }
 
-func TestReconcilePropagateLabels(t *testing.T) {
+func TestReconcilePropagateLabelsAndAnnotations(t *testing.T) {
 	names.TestingSeed()
 
 	ps := []*v1beta1.Pipeline{simpleHelloWorldPipeline}
-	prs := []*v1beta1.PipelineRun{{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "test-pipeline-run-with-labels",
-			Namespace: "foo",
-			Labels: map[string]string{
-				"PipelineRunLabel":        "PipelineRunValue",
-				pipeline.PipelineLabelKey: "WillNotBeUsed",
-			},
-		},
-		Spec: v1beta1.PipelineRunSpec{
-			PipelineRef:        &v1beta1.PipelineRef{Name: "test-pipeline"},
-			ServiceAccountName: "test-sa",
-		},
-	}}
+	prs := []*v1beta1.PipelineRun{parse.MustParsePipelineRun(t, `
+metadata:
+  annotations:
+    PipelineRunAnnotation: PipelineRunValue
+  labels:
+    PipelineRunLabel: PipelineRunValue
+    tekton.dev/pipeline: WillNotBeUsed
+  name: test-pipeline-run-with-labels
+  namespace: foo
+spec:
+  pipelineRef:
+    name: test-pipeline
+  serviceAccountName: test-sa
+`)}
 	ts := []*v1beta1.Task{simpleHelloWorldTask}
 
-	expectedObjectMeta := taskRunObjectMeta("test-pipeline-run-with-labels-hello-world-1", "foo", "test-pipeline-run-with-labels", "test-pipeline", "hello-world-1", false)
-	expectedObjectMeta.Labels["PipelineRunLabel"] = "PipelineRunValue"
-	expected := &v1beta1.TaskRun{
-		ObjectMeta: expectedObjectMeta,
-		Spec: v1beta1.TaskRunSpec{
-			TaskRef: &v1beta1.TaskRef{
-				Name: "hello-world",
-			},
-			ServiceAccountName: "test-sa",
-			Resources:          &v1beta1.TaskRunResources{},
-			Timeout:            &metav1.Duration{Duration: config.DefaultTimeoutMinutes * time.Minute},
-		},
-	}
+	expected := parse.MustParseTaskRun(t, `
+metadata:
```

Comment:
Once #4749 lands, I'll modify this to use the `ObjectMeta` injection logic I added there, which will clean this particular YAML chunk up.